lib/logstash/inputs/beats.rb

```diff
@@ -189,6 +205,17 @@ def register
       @logger.warn("configured ssl_key => #{@ssl_key.inspect} will not be used") if @ssl_key
     end
 
+    @enrich = resolve_enriches
+
+    @include_codec_tag = original_params.include?('include_codec_tag') ? params['include_codec_tag'] : include_codec_metadata?
+    @ssl_peer_metadata = original_params.include?('ssl_peer_metadata') ? params['ssl_peer_metadata'] : @enrich.include?('ssl_peer_metadata')
+
```

Comment:
It seems we may need to warn the user when `enrich` is alone defined but codec is missed.

```suggestion
if !original_params.include?('codec') && @enrich && @enrich.include?('codec_metadata')
      @logger.warn("`enrich` configuration is ignored since `codec` isn't provided. You can add codec definition in the setting to include codec metadata.")
    end
```